src/mjolnir/landmarks.cc

```diff
@@ -296,6 +318,226 @@ bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+                       const std::vector<GraphId>& tileset,
+                       const size_t& thread_number,
+                       const size_t& total_threads,
+                       std::promise<std::string>& seq_file_name) {
+  // Open the database and create a graph reader
+  const std::string dbname = pt.get_child("mjolnir").get<std::string>("landmarks", "");
+
+  LandmarkDatabase db(dbname, true);
+  GraphReader reader(pt.get_child("mjolnir"));
+  // create the sequence file
+  std::string file_name = "landmark_dump_" + std::to_string(thread_number);
+  midgard::sequence<std::pair<GraphId, uint64_t>> seq_file(file_name, true);
+
+  for (size_t i = 0; i < tileset.size(); ++i) {
+    // assign tiles for threads
+    if (i % total_threads == thread_number) {
+      // get landmarks in the tile
+      midgard::AABB2<PointLL> bbox = baldr::TileHierarchy::GetGraphIdBoundingBox(tileset[i]);
+
+      std::vector<Landmark> landmarks =
+          db.get_landmarks_by_bbox(bbox.minx(), bbox.miny(), bbox.maxx(), bbox.maxy());
+
+      // find and collect all nearby path locations for the landmarks
+      for (const auto& landmark : landmarks) {
+        baldr::Location landmark_location(midgard::PointLL{landmark.lng, landmark.lat},
+                                          baldr::Location::StopType::BREAK, 0, 0,
+                                          25); // a 25m radius, which allows us to only keep the close
+                                               // ones in the tight cities
+        landmark_location.search_cutoff_ = 75.; // a 75m search cutoff should allow us to get gas
+                                                // stations that are off the road a bit (for parking)
+
+        // call loki::Search to get nearby edges to each landmark
+        std::unordered_map<valhalla::baldr::Location, PathLocation> result =
+            loki::Search({landmark_location}, reader, sif::CreateNoCost({}));
+        if (result.size() > 1) {
+          throw std::logic_error("Error occurred in calling loki::Search. Result size is " +
+                                 std::to_string(result.size()) + ", but should be one or zero");
+        }
+        if (result.size() == 0) {
+          continue;
+        }
+
+        std::vector<PathLocation::PathEdge> edges = result.begin()->second.edges;
+        // for each edge insert edgeid - landmark_pkey pair into the sequence file
+        // TODO: maybe do some filtering and only keep some of the edges it finds?
+        for (const auto& edge : edges) {
+          seq_file.push_back(std::make_pair(edge.id, landmark.id));
+        }
+      }
+    }
+  }
+  // sort the sequence by tiles
+  seq_file.sort(sort_seq_file);
+
+  seq_file_name.set_value(file_name);
+}
+
+// Update tiles to associate landmarks, return some stats about the numbers of updated tiles, edges,
+// and landmarks
+void UpdateTiles(midgard::sequence<std::pair<GraphId, uint64_t>>& seq_file,
```

Comment:
we should mention that seq_file is by ref but should not be modified by these threads. the other option is to pass teh file name and have each thread open the same file